### ISSUE TITLE
Show hint of resolution source

### ISSUE THREAD

**Issue by kuon** [ogham/dog#7]
One killer feature would be to have a hint on the resolution source. For example, if I do `dog example.com` and I have `example.com`remmaped in my `/etc/hosts` file, it should show something, like a `H` at the end of the line.

I don't know how feasible it is, but having a distinction between the following sources would be great:

- Regular DNS
- mDNS (local)
- `/etc/hosts`

I think the last one could be easy, just grep the file, but I don't know about the first two.

**Comment by ogham** [comment]
Having dog print a message or a flag is definitely something I could provide. I'm pretty sure I've made that mistake before as well!

To be clear (as I misunderstood this issue the first time I read it): I don't want to adapt dog so it reads the _addresses_ from the hosts file and returns those instead. I want it to stay a DNS client, not a general address resolver. But alerting the user that the results may be different from what they expect, keeps it within its purpose.

> I think the last one could be easy, just grep the file, but I don't know about the first two.

I think the same — dog already parses `/etc/resolv.conf`, so adding a hosts file parser isn't too far removed from its wheelhouse. I'm not sure how to separate mDNS results from regular ones.

**Comment by kuon** [comment]
Yes, exactly, dog should not do anything smart to resolve the address in a different way than the system, but if dog can just alert the user that he has an override in `/etc/hosts` or that some misconfigured (I look at you netgear) or unexpected anycast DNS provider did override the response.

**Comment by ogham** [comment]
I was able to do the hosts file checking without too much fuss:

![Screenshot 2021-04-04 at 17 43 05](https://user-images.githubusercontent.com/503760/113515578-3817f480-956d-11eb-8e66-af6578e39c72.png)
